x-pack/test/alerting_api_integration/basic/tests/actions/builtin_action_types/slack.ts

```diff
@@ -4,26 +4,23 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import http from 'http';
 import { FtrProviderContext } from '../../../../common/ftr_provider_context';
 
-import {
-  getExternalServiceSimulatorPath,
-  ExternalServiceSimulator,
-} from '../../../../common/fixtures/plugins/actions_simulators/server/plugin';
+import { getSlackServer } from '../../../../common/fixtures/plugins/actions_simulators/server/plugin';
 
 // eslint-disable-next-line import/no-default-export
 export default function slackTest({ getService }: FtrProviderContext) {
   const supertest = getService('supertest');
-  const kibanaServer = getService('kibanaServer');
 
   describe('slack action', () => {
-    let slackSimulatorURL: string = '<could not determine kibana url>';
+    let slackSimulatorURL: string = '';
+    let slackServer: http.Server;
 
-    // need to wait for kibanaServer to settle ...
-    before(() => {
-      slackSimulatorURL = kibanaServer.resolveUrl(
-        getExternalServiceSimulatorPath(ExternalServiceSimulator.SLACK)
-      );
+    before(async () => {
+      slackServer = await getSlackServer();
+      slackServer.listen(9005);
```

Comment:
I'm a little worried about the number of ports we are using - 9001 through 9005.  It's possible that someone may have a local server running on those ports, which would likely cause either that local server or the test server to be inaccessible or break in some way.

Seems like we could _probably_ run all these servers on the same port, and then use the first path of the URL to gate which "virtual" server it was going to; eg, `http://localhost:9001/slack-1` would be map to one the existing slack servers, and `http://localhost:9001/webhook-1` would map to one of the existing webhook servers, etc.  

That would at least get us down to just using one port.

The other thing we could is try to let node / the tcp stack pick the port for us, presumably from an unused port.  You can use a package like `https://www.npmjs.com/package/get-port` to request an unused port, then use it when creating the server and urls to the server.  In practice, I've often had issues using this technique, as it occasionally has issues, or sometimes the libraries do bad things like "test" the socket before using it (which they shouldn't).

Probably at the very least, if we leave them as hard-coded numbers like this, we should make them exported `const` values somewhere they can be easily changed.